src/App.js

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+
+import { TodoList } from "./components/TodoList";
+import TodoForm from "./components/TodoForm";
+
+import "./styles.scss";
+
+export default function App() {
+  const [todos, setTodos] = useState([
+    { id: 1, text: "Buy milk", done: true },
+    { id: 2, text: "Buy bread", done: false },
+  ]);
+
+  function addTodo(todo) {
+    setTodos([
+      ...todos,
+      { ...todo,
+        id: todos.length > 0 ? Math.max(...todos.map(val => val.id)) + 1 : 1 }
+    ])
+  }
+
+  function toggleTodo(todo) {
+    const newTodos = [...todos].map(i => i.id === todo.id ? {...i, done: !i.done } : {...i});
+    setTodos(newTodos);
+  }
+
+  function deleteTodo(todo) {
+    setTodos(prevTodos => prevTodos.filter(task => task.id !== todo.id))
+  }
+
+  function editTodo(todo) {
+    const newTodos = [...todos].map(i => i.id === todo.id ?
+        { ...i, text: prompt('Enter your updated todo text')} : {...i}
```

Comment:
Using prompt is an interesting choice. While this works, there's no handling for the user clicking "Cancel" on the prompt, resulting in a null return which clears the current text of the task.
We also would have preferred to see deeper consideration for the UX; for example, rather than using a prompt, we'd suggest editing the current text in an input field on the todo list item.

Reply:
You're right, I guess I should have added: `{...i, text: prompt('Enter your updated todo text')) || 'no text')} : {...i}` - something like this, I haven't tested this code that I wrote in the comment. But yeah, essentially, a truthy check done one way or another.